http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java

```diff
@@ -138,6 +150,34 @@ private HttpProxyOptions buildProxyOptions(ProxyConfiguration proxyConfiguration
         return clientProxyOptions;
     }
 
+    private SocketOptions buildSocketOptions(DefaultBuilder builder, AttributeMap config) {
+        SocketOptions clientSocketOptions = new SocketOptions();
+
+        Duration connectionTimeout = config.get(SdkHttpConfigurationOption.CONNECTION_TIMEOUT);
+        if (connectionTimeout != null) {
+            clientSocketOptions.connectTimeoutMs = (int) Long.min(connectionTimeout.toMillis(), Integer.MAX_VALUE);
```

Comment:
`NumericUtils.saturatedCast(connectionTimeout.toMillis())` can be used instead
(same for `keepAliveIntervalSecs` and `keepAliveTimeoutSecs` L166 and L167)